parser.mly

```diff
@@ -25,15 +25,15 @@ let scope_annotation (mode, xs) =
 %%
 
 program:
-| prog=list(instruction_line) EOF
+| list(NEWLINE) prog=list(instruction_line) EOF
   {
     let annotations, instructions = List.split prog in
     (Array.of_list instructions,
      Array.of_list annotations)
   }
 
 instruction_line:
-| a=scope_annotation i=instruction NEWLINE { (a, i) }
+| a=scope_annotation i=instruction nonempty_list(NEWLINE) { (a, i) }
```

Comment:
I think that we could even consider scopes as whitespace and ignore them wherever, allowing
`branch x l1 l2 z <- (x + 1)` on a single line -- in particular the fact that primops are parenthesized makes our grammar highly unambiguous. The question is what we want.

One thing that we could do is introduce a semicolon to explicitly conclude instructions, and then allow newlines everywhere (for example `branch (x == x2)\nl1 l2`. Or we could allow to escape newlines to explicitly continue on the next line, à la Python. Or just allow newlines between scopes and instructions, as you suggest.